internal/bot/bot.go

```diff
@@ -40,6 +40,14 @@ func (b *Bot) Answer(receiveMsg *messages.Receive) ([]query.Answer, error) {
 
 	previousState := machine.State
 
+	// Set existing conversation to false if in initial state.
+	// This tells the bot that it is having a new conversation
+	if initialID, ok := b.Domain.StateTable["initial"]; ok {
+		if initialID == machine.State {
+			isExistingConversation = false
+		}
+	}
+
```

Comment:
Actually "initial" is just a name for the state number 0, therefore I think this could be simplified to:

```go
	isExistingConversation = !(machine.State == 0)
```

Reply:
`0` is not very human/programmer friendly maybe we can have a constant in the `FSM` package we can reference as the initial state of `0`. Similar to how the `http` package does it with HTTP return codes.

```go
const (
    StatusContinue           = 100 // RFC 7231, 6.2.1
    StatusSwitchingProtocols = 101 // RFC 7231, 6.2.2
    StatusProcessing         = 102 // RFC 2518, 10.1
    StatusEarlyHints         = 103 // RFC 8297

    StatusOK                   = 200 // RFC 7231, 6.3.1
    StatusCreated              = 201 // RFC 7231, 6.3.2
)
```

So in the `FSM` it could be...

```go
const (
  StateInitial = 0
  StateAny     = -1
)
```

Reply:
I totally agree, those two constant states are more friendly and also remove some "hardcoded" numbers.

Reply:
Ok I will make that change